build/steps-build-ios.yml

```diff
@@ -0,0 +1,99 @@
+parameters:
+- name: iosCertificateFile
+  type: string
+  default: ''
+- name: iosCertificatePassword
+  type: string
+  default: ''
+- name: iosProvisioningProfileFile
+  type: string
+  default: ''
+- name: pathToSrc
+  type: string
+  default: ''
+- name: solutionName
+  type: string
+  default: ''
+- name: pathToInfoPlist
+  type: string
+  default: ''
+
+steps:
+- template: templates/dotnet-install.yml
+
+#-if false
+- download: current
+  condition: eq('GeneratedApp', '${{ parameters.solutionName }}')
+  artifact: GeneratedApp
+#-endif
+
+- template: templates/gitversion.yml
+
+- task: InstallAppleCertificate@2
+  displayName: Install Certificate
+  inputs:
+    certSecureFile: ${{ parameters.iosCertificateFile }}
+    certPwd: $(AppleCertificatePassword)
+    keychain: temp
+    deleteCert: true
+
+- task: InstallAppleProvisioningProfile@1
+  displayName: 'Install Apple Provisioning Profile'
+  name: provisioningProfile
+  inputs:
+    provisioningProfileLocation: 'secureFiles'
+    provProfileSecureFile: ${{ parameters.iosProvisioningProfileFile }}
+
+- task: UpdateiOSVersionInfoPlist@1
+  displayName: "Bump iOS Version"
+  inputs:
+    infoPlistPath: ${{ parameters.pathToInfoPlist }}
+    bundleShortVersionString: '$(MajorMinorPatch)'
+    bundleVersion: '$(PreReleaseNumber)' 
+
+- task: ios-bundle-identifier@1
+  displayName: "Replace iOS Bundle Identifier"
+  inputs:
+    sourcePath: ${{ parameters.pathToInfoPlist }}
+    bundleIdentifier: $(ApplicationIdentifier)
+    printFile: true
+
+- bash: |
+    cd ${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile
+    dotnet $(dotNetCommand) -f:$(BuildTargetFramework) -c:Release /p:ApplicationIdentifier=$(ApplicationIdentifier) /p:CodesignProvision=$(provisioningProfile.provisioningProfileUuid) /p:ArchiveOnBuild=true /bl:$(build.ArtifactStagingDirectory)/build-$(BuildTargetFramework).binlog "/p:InformationalVersion=$GITVERSION_INFORMATIONALVERSION" /p:IsLightBuild=$(IsLightBuild)
+  displayName: Build project for Release - $(BuildTargetFramework)
+
+- task: CopyFiles@2
+  displayName: 'Copy IPA'
+  inputs:
+    Contents: '**/*.ipa'
+    flattenFolders : true
+    SourceFolder: '${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile/bin/Release/net6.0-ios'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)'
+
+- task: CopyFiles@2
+  displayName: 'Copy DSYM'
+  inputs:
+    Contents: '**/*.dSYM/**'
+    flattenFolders : true
+    SourceFolder: '${{ parameters.pathToSrc }}/app/${{ parameters.solutionName }}.Mobile/bin/Release/net6.0-ios'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)/${{ parameters.solutionName }}.Mobile.dSYM'
+
+- publish: $(Build.ArtifactStagingDirectory)
+  displayName: 'Publish Artifact $(ApplicationConfiguration) | $(ApplicationPlatform)'
```

Comment:
You don't pass in the `ApplicationConfiguration `leading in improper display in  the CI. 
You should pass it form the AzurePipelines.yml depending on the stage (Production etc ..) 

Same thing for `steps-build-android`

![image](https://user-images.githubusercontent.com/15191066/206743878-2598a152-cdc4-4a40-8c56-5d040683ff58.png)

Reply:
I'm removing it for now since I wanted to do another Pull Request for cleaning and adding some variables like this.